packages/x-date-pickers/src/internals/hooks/usePickerState.ts

```diff
@@ -141,14 +160,24 @@ export const usePickerState = <TInput, TDateValue>(
     [acceptDate, disableCloseOnSelect, isMobileKeyboardViewOpen, draftState.draft],
   );
 
+  const handleInputChange = React.useCallback(
+    (date: TDateValue, keyboardInputValue?: string) => {
+      const cleanDate = valueManager.mergeOldAndNewValues
+        ? valueManager.mergeOldAndNewValues(utils, lastValidDateValue, date)
+        : date;
+      onChange(cleanDate, keyboardInputValue);
+    },
```

Comment:
If `valueManager` can be provided by developers, maybe renaming `prevValue` by `prevValidValue` to be more accurate on what is done internally

Reply:
It is fully internal